worker/src/lib.rs

```diff
@@ -376,6 +376,95 @@ pub async fn dns_lookup(
     })
 }
 
+#[wasm_bindgen]
+pub async fn gist_lookup(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    gist_id: String,
+    github_username: String,
+) -> Promise {
+    initialize_logging();
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+
+        let gist_result = jserr!(github::retrieve_gist_message(gist_id.clone()).await);
+
+        let mut vc = jserr!(github::build_gist_vc(&pk, github_username.clone()));
+
+        // check for matching usernames
+        if github_username.to_lowercase() != gist_result.owner.login.to_lowercase() {
+            jserr!(Err(anyhow!(format!(
+                "Different Github username {} v. {}",
+                github_username, gist_result.owner.login
+            ))));
+        }
+
+        let (sig_target, sig) = jserr!(extract_signature(
+            gist_result.files.tzprofiles_verification.content.clone()
+        ));
+
+        jserr!(verify_signature(&sig_target, &pk, &sig));
+
+        let mut props = HashMap::new();
+        props.insert(
+            "publicKeyJwk".to_string(),
+            jserr!(serde_json::to_value(pk.clone())),
+        );
+
+        let mut evidence_map = HashMap::new();
+
+        evidence_map.insert(
+            "timestamp".to_string(),
+            serde_json::Value::String(Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true)),
+        );
+
+        evidence_map.insert(
+            "gistId".to_string(),
+            serde_json::Value::String(gist_id.to_string()),
+        );
```

Comment:
There's the version id/sha in the response as well, under `history[-1].version`

Reply:
Added to evidence map